tests/fail-dep/concurrency/libc_pthread_mutex_move.rs

```diff
@@ -0,0 +1,72 @@
+//@ignore-target-windows: No pthreads on Windows
+//@[unlock_register]only-target-linux: recursive initializers are non-standard.
+//@revisions: lock trylock unlock_register unlock_detect init
```

Comment:
Please declare the revisions before using them.

Also, I am confused by the names. `lock` really is `static_initializer`, right? `init` is `initializer_function`. And why is `unlock_register` only for Linux? Recursive mutexes exist on all platforms, they just don't have a static initializer on non-Linux.

Maybe it just needs comments explaining what these variants are testing, but currently I am quite confused. I expected only 2 tests -- one that uses the static initializer and one that uses `pthread_mutex_init`.

Reply:
Regarding the tests:

In my mind what needs to be tested is the ability of the `pthread_mutex_*` functions to record and then later detect a moved `pthread_mutex_t`. So for each such function I call it two times. If the move is caught it means that the function both recorded the original address and detected the move. At least that's the intention.

For the `unlock` function however I needed a slight different approach. Static initializers are only supported on linux, and `unlock` can only be called on recursive mutexes if they are unlocked. So I used two different revisions, one to test that `unlock` correctly registers the address (with a static initializer), and one to test that it detects a move, when called after init.

Reply:
I see, thanks.

However, given our implementation, we know this is all the same codepath, so it doesn't seem worth testing *all* combinations.

Reply:
ack, removed the extra tests.